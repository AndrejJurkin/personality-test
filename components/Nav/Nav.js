import React from "react";
import Button from "../Button/Button";
import navLinks from "../../resources/navLinks";
import clsx from "clsx";
import Link from "next/link";

export default function Nav({ dark, showDivider, offsetTop }) {
  return (
    <>
      <nav
        className={clsx(
          dark && "dark",
          showDivider && "border-b border-gray-200",
          "absolute top-0 left-0 z-10 h-20 w-full px-4 text-white md:px-12 lg:px-24"
        )}
      >
        <div
          className={clsx(
            "mx-auto flex h-full max-w-5xl flex-row items-center justify-between",
            "dark:text-slate-800"
          )}
        >
          <Link href="/" data-testid="app-logo">
            <a className="font-display text-xl">Quiso</a>
          </Link>
          <ul className="flex hidden flex-row gap-12 text-sm md:flex">
            {navLinks.map((link, index) => (
              <li key={index}>{link.name}</li>
            ))}
          </ul>
          <Link href="/test" passHref>
            <a>
              <Button primary>Take the Test</Button>
            </a>
          </Link>
        </div>
      </nav>
      {offsetTop && <div className="h-24" />}
    </>
  );
}
