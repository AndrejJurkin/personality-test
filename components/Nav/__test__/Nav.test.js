import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Nav from "../Nav";
import navLinks from "../../../resources/navLinks";

describe("Nav", () => {
  it("renders navigation role", () => {
    const { getByRole } = render(<Nav />);
    expect(getByRole("navigation")).toBeInTheDocument();
  });

  it("renders nav links", () => {
    const { getByText } = render(<Nav />);
    navLinks.forEach((link) => {
      expect(getByText(link.name)).toBeInTheDocument();
    });
  });

  it("renders CTA", () => {
    const { getByText } = render(<Nav />);
    expect(getByText("Take the Test")).toBeInTheDocument();
  });

  it("renders component unchanged", () => {
    const { container } = render(<Nav />);
    expect(container).toMatchSnapshot();
  });
});
