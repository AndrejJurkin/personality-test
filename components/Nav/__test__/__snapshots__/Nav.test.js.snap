// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders component unchanged 1`] = `
<div>
  <nav
    class="absolute top-0 left-0 z-10 h-20 w-full px-4 text-white md:px-12 lg:px-24"
  >
    <div
      class="mx-auto flex h-full max-w-5xl flex-row items-center justify-between dark:text-slate-800"
    >
      <a
        class="font-display text-xl"
        href="/"
      >
        Quiso
      </a>
      <ul
        class="flex hidden flex-row gap-12 text-sm md:flex"
      >
        <li>
          Home
        </li>
        <li>
          About
        </li>
        <li>
          Contact
        </li>
      </ul>
      <a
        href="/test"
      >
        <button
          class="button button--primary button--medium"
        >
          Take the Test
        </button>
      </a>
    </div>
  </nav>
</div>
`;
